src/main/java/frc/robot/subsystems/vision/VisionIOReal.java

```diff
@@ -1,9 +1,123 @@
+            rightCamFromRobot);
+
+        // Left camera and object that estimates robot pose from left cam input
+        leftCam = new PhotonCamera(frontCamName);
+        leftCamPoseEstimator = new PhotonPoseEstimator(
+            aprilTagFieldLayout,
+            MULTI_TAG_PNP_ON_COPROCESSOR,
+            leftCam,
+            leftCamFromRobot);
+    }
+
+    @Override
+    public void updateInputs(VisionIOInputs inputs) {
+
+        updateEstimatedPose();
+        updateVisibleToteAprilTags();
+
+        inputs.estimatedRobotPose = estimatedRobotPose;
+        inputs.visibleToteAprilTags = visibleToteAprilTags;
+    }
+
+    private Pose3d getUpdatedIndiviualCameraEstimatedPose(PhotonPoseEstimator poseEstimator) {
+        // If the pose estimator has a value then return the pose
+        try {
+            return poseEstimator.update().get().estimatedPose;
+        // Otherwise, return the current pose of the robot that is estimated by vision
+        // In practice, this should have the effect of valuing estimated poses more the more camera see tags
+        } catch(NoSuchElementException e) {
+            return estimatedRobotPose;
+        }
+
+    }
+
+    public void updateEstimatedPose() {
+
+        // Initialize a temporary pose
+        Pose3d averagePose = new Pose3d();
+
+        // Add all three poses together
+        averagePose = averagePose.plus(new Transform3d(new Pose3d(), getUpdatedIndiviualCameraEstimatedPose(frontCamPoseEstimator)));
+        averagePose = averagePose.plus(new Transform3d(new Pose3d(), getUpdatedIndiviualCameraEstimatedPose(rightCamPoseEstimator)));
+        averagePose = averagePose.plus(new Transform3d(new Pose3d(), getUpdatedIndiviualCameraEstimatedPose(leftCamPoseEstimator)));
+        // Divide by number of poses (3) to get the average pose
+        averagePose = averagePose.div(3);
+
+        estimatedRobotPose = averagePose;
+
+     }
+
+     private Transform3d[] getUpdatedIndividualCameraVisibleToteAprilTags(PhotonCamera camera, Transform3d fromRobot, Transform3d[] curList) {
+        Transform3d[] localVisibleToteAprilTags = curList;
+        // For every visible target...
+        for(PhotonTrackedTarget target : camera.getLatestResult().getTargets()) {
+            // If its ID is less than 12...
+            if (target.getFiducialId() < 12) {
+                localVisibleToteAprilTags[target.getFiducialId()] = 
+                // If there isnt already a value for that april tag...
+                localVisibleToteAprilTags[target.getFiducialId()] == null ? 
+                // Set the index of its ID to the transform from the bot to the target
+                target.getBestCameraToTarget().plus(fromRobot) : 
+                // Otherwise set the index of its ID to the average of the value that is there and the transform from the bot to the target
+                localVisibleToteAprilTags[target.getFiducialId()].plus(target.getBestCameraToTarget().plus(fromRobot)).div(2);
```

Comment:
In general, the amount that you trust one measurement or another should be scaled by the uncertainty of the measurement (could be the distance away from the tag, among other things). However, due to the layout of the cameras on our current robot it's impossible for two cameras to see the same april tag (unless you have a mirror...hmmm) so it's not really an issue here.

Reply:
im pretty sure it is possible, though unlikely, for just two cameras to see the same april tag, although it is definitely impossible for all three to see the same april tag. I will admit that use for this averaging in this case is pretty redundant, as the distance required to see one april tag on multiple cameras is way farther than we would want to be using it at, so in practice any april tags we would use this for would only be on one camera